cobalt-ast/src/cfg.rs

```diff
@@ -732,19 +883,31 @@ impl<'a, 'ctx> Cfg<'a, 'ctx> {
                     blk = Some(n)
                 }
             }
+
             let true_ = ctx.context.bool_type().const_all_ones();
             let false_ = ctx.context.bool_type().const_zero();
+
             let blk = if let Some(blk) = blk {
                 blk
             } else {
+                // We are checking if the value has moved before an instruction which does not even
+                // belong to the same block as the value, so it doesn't make sense to compare them.
+                // But for convention we will return false.
                 return false_;
             };
+
+            // - 1: Starting from the last move and going backwards...
+            // - 2: If the move is wrt this variable and happens before the specified
+            // instruction...
+            // - 3: If the move is a use, then yes the variable has moved before the instruction.
+            // If the move was a store then the variable hasn't moved.
```

Comment:
The phrasing here could be a bit clearer:
- It checks if the *use* is a *move*
- If there was a store there (which is *not* a use, but I forgot to change some variable names), then it is guaranteed that the variable is initialized (not moved) *after* this block.